client/consensus/grandpa/aux_schema.go

```diff
@@ -179,15 +179,39 @@ func UpdateAuthoritySet[H comparable, N constraints.Unsigned](store AuxStore, se
 // We always keep around the justification for the best finalized block and overwrite it
 // as we finalize new blocks, this makes sure that we don't store useless justifications
 // but can always prove finality of the latest block.
-func UpdateBestJustification() {
-	// TODO impl when we have justification logic
-	panic("impl")
+func UpdateBestJustification[H HashI, N constraints.Unsigned, S comparable, ID constraints.Ordered, Header HeaderI[H, N]](store AuxStore, justification Justification[H, N, S, ID, Header]) error {
+	encodedJustificaiton, err := scale.Marshal(justification)
+	if err != nil {
+		return err
+	}
+
+	insert := []KeyValue{
+		{BEST_JUSTIFICATION, encodedJustificaiton},
+	}
+	err = store.Insert(insert, nil)
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
 // BestJustification  Fetch the justification for the latest block finalized by GRANDPA, if any.
-func BestJustification() {
-	// TODO impl when we have justification logic
-	panic("impl")
+func BestJustification[H HashI, N constraints.Unsigned, S comparable, ID constraints.Ordered, Header HeaderI[H, N]](store AuxStore) (*Justification[H, N, S, ID, Header], error) {
+	encodedJustification, err := store.Get(BEST_JUSTIFICATION)
+	if err != nil {
+		return nil, err
+	}
+
+	if encodedJustification != nil {
+		justification := Justification[H, N, S, ID, Header]{}
+		err = scale.Unmarshal(*encodedJustification, &justification)
+		if err != nil {
+			return nil, err
+		}
+		return &justification, nil
+	}
+	// Can return error here is we want to error out if nothing returned
+	return nil, nil
```

Comment:
I would suggest returning a sentinel error in case of not found, and also returning early 
```suggestion
        if encodedJustification == nil {
		return nil, ErrJustificationNotFound
        }
        
	justification := Justification[H, N, S, ID, Header]{}
	err = scale.Unmarshal(*encodedJustification, &justification)
	if err != nil {
		return nil, fmt.Errof("unmarshalling: %w", err)
	}
	return &justification, nil
```